src/appleseed/renderer/modeling/surfaceshader/constantsurfaceshader.cpp

```diff
@@ -117,18 +117,15 @@ namespace
                 &values);
 
             // Initialize the shading result.
-            shading_result.m_color_space = ColorSpaceSpectral;
-            shading_result.m_main.m_color = values.m_color;
-            shading_result.m_aovs.m_color.set(0.0f);
-            shading_result.m_aovs.m_alpha.set(0.0f);
+            aov_accumulators.beauty().set(values.m_color);
 
             // This surface shader can override alpha.
             if (m_alpha_source == AlphaSourceColor)
-                shading_result.m_main.m_alpha = values.m_alpha;
+                aov_accumulators.alpha().set(values.m_alpha);
 
             // Apply multipliers.
-            shading_result.m_main.m_color *= values.m_color_multiplier;
-            shading_result.m_main.m_alpha *= values.m_alpha_multiplier;
+            aov_accumulators.beauty().apply_multiplier(values.m_color_multiplier);
+            aov_accumulators.alpha().apply_multiplier(Alpha(values.m_alpha_multiplier));
```

Comment:
You misunderstood, I'm talking about the cast to `Alpha`. It seems that `apply_multiplier()` takes a float argument, so there really is no need for that cast.